src/https-server.ts

```diff
@@ -0,0 +1,186 @@
+jlWuikEShaLSM5ZumeRQHGID+AuAcEXixNTc/pkQ7IDQs++jwonV4oO4JofeAxdd
+Chqh0O0XKDKrpI0FxJovBkA8Ssp/AEQjhuJu0tXdsMGnloFAb9tTyhp3zpOUIA1V
+0A27BN2psEOle5m4nWIftrA0sDIvzvIQYFY3jFoHIUoGiBorBCdPQyA6r7ivnFT/
+yQQsHK+aXOW4og5lcfygjkGTZpTQXWnnbsybfAW9dkRSNj6mzXTiM+VTF1sdEscQ
+Au+f9rUsa+k7uZxXT9pB1S3vr98Gkn+M8SGBFLYPctPdXnVSOeWazVA2Kj0qsbuE
+HlVD72TaquczxQlt9nfRE7oTn3s9p+BVqZfk1zSFpvxqoDmhzif+EmkPsnsRY1QM
+CVhXWHnSdo2/xiWo07Bblz8Ixv5Dut5b5nPG8YELRGLTyhgVMMHHGmVz5ZwtjV0f
+1sTZhDBgSQmJzbSQ7oRSL71FMnVZYdKo0PVJlIy3kPV1VW8GLN3xTXHvIUAc/gvF
+yLzGNXRjf9++Xcw993mRbzmDXFpHbgyh3H+K0DkruTh9iSArMQKCAQEA+yDC6OGB
+BBATYZZaRw/OlOE3wpBB/2dfyst92nAadHis1ygtjG6WN8D381UhBrr2Ime6GGQO
+/kXFdJ4aBZQIc8gRFUVGw1jzYKGU/iw56kdwYxYM4nbKVnvWvPpoxdZh+2rOLdTc
+sGugp10sS35t+1zy90jUIvKMdkhNDYCvdx6Ca4QwYqiXfdZom04HYyDSDP7ML+Xy
+bBUUJVwv2w/ylN1mtPw/mONXU1Ua0dHry91a05b6s0D9aq4X3o0HMIPSmX/wvH9Z
+U0VXlpndkiZIbRFzVaupBGUghrKRqcRER9SF/CWBwBVqgANPkt3ogNwNBZMjgC7/
+WN9TCabvlvs6LQKCAQEA3n5ppTAwTIBvh2CrgjK5UJbJNWi0voPsRmnK7ubyxSS5
+JTC4jfP60KdcRF6cMlCJIcNMLkONwADEAZJa+8sGCQ2dBemLJd7dSwh3Y/kjGEp/
+VOQC9YN8fR30OX/Wl3RObDp5/zsMLv8mAQVjUIIXth6de7Min7PnXxSy2Jl5N60h
+AzwGHtVX+KP6JoqkcWKgnoZV+V56EfzJ9uAyifFzQymT/KEogU5peeywtf/lZb43
+09PZqT/pyDkCYZv5QWEWODvm7lM8xhNi0Aq/n6h/D5q2NYwN8cd+eteJnaL4OMwe
++5mi202Y+E73OzMuvRQDjzCqeFzi+WOqET+rrfMdCwKCAQA5kdkc6hVZiRXoWT94
+6vbAyqCtycRXeKtNIjlHyktGSCPXby6B8xgcz/AHqIxZrqKeYIkY7ldBk10gzt3I
+SZdmc4b3CVqfjdS9Yk11t03Pl3D8RiHOCM155gnkaMN8Fxl/OYzqlNHC3+mEdKei
+5ujvpu9MITYAjPoGY6zGW6nd9PCRFUzCUUPpZS217L1kzDFbKU5F7MEWZf6oUenm
+nlFOLzNMakdqFTP/tpLfZvBXW/dmakYuscdmnWJ7cnkjdhThe6MEk2022b6ql7tj
+tcMEvetiyGeGymgaePVLpeLBVMcidra4AjDYB+pGEPksEUYq8QBCcoO3K/stS0fB
+pPZBAoIBABzQFBBPUEHn9Al9DFjdBsCX9lCTUazGWse3lY8gm1nKaquVRqEjclyo
+/7re4a1mXovfxbLMDsRJlRh7AotO/GYZvDbyypFJ7jB0VTm+HrxbkWfF61guAeYW
+BfWYR4rvheRgIkJaPDat1hcVwom/fyCXkF9fkGFMI+9dN9fy8A60vey0TjqZ+D17
+xr5NAASq+A/vZ5zWjkpcCS+OUx0oBnwxGzgBCsTMCb6/VxcKn4NwgbxN2Ach/r1u
+HI/Vag78aqs3nq1IXDtucUJt0ej0Ihg8OdJvY7Lm2ctAtFKUsz5vuKdueveZcfDe
+EQg6lZfoypk1fS3/IWmYbYK0UyBAtocCggEAWMPAdj6HlhnM77TVmwJm/pfyj5oB
+Z6gTZaNrYmsIQJVdkC0FEJuEpgQsM28TlCz3nmPuZw2A8nRXWFWkHbEw3NAgGkqV
+N5GGMuRbFizzAcG6yg9+KojU+qjsGexeVJhsuONH/P2THUvsoX7kzLXKq1IeMLy3
+vwNaL/+dFfJ8cmi9FYaadRqQ59RZZg7J5KK80fFyzsat1LWUIXHsEYdZ0T0hip2r
+r7Iao7vY/DxzVeMOze2yO5OOOgbq8OG5HkN6UnD/PhevUbADm19khunQgTRdKZAI
+ng8OrBbXN/3vyK6QseIUz0K5mMjhzEBeFF/3o2yxnJJtz2q17Y/Ck9qshQ==
+-----END RSA PRIVATE KEY-----`
+
+export type HttpsServerOptions = https.ServerOptions
+
+export class HttpsServer extends HttpServerBase<https.Server> {
+  public cert: HttpsServerOptions['cert']
+  public key: HttpsServerOptions['key']
+  public caAgent: https.Agent
+  public clientCertAgent: https.Agent
+
+  constructor(options: HttpsServerOptions, requestListener: HttpRequestListener) {
+    options.cert
+    options = {
+      cert: localhostCertificate,
+      key: localhostKey,
+      ca: clientCaCertificate,
+      requestCert: true,
+      rejectUnauthorized: false, // so we can do own error handling
```

Comment:
What happens if the developer doesn't add any error handling? Wouldn't it just let anything through?

Reply:
Yes, I was thinking the same. But my reasoning is that we are not trying to replace "express" or be a real http server, so unsafe defaults is ok if it makes testing easier and in this case, it makes the HttpsServer work both with and without client certs by default.

Reply:
Maybe it's the best default for ease of use. I'm just worried that we'll end up in a situation where someone uses this default setup and assumes that the SSL works while it actually doesn't.